docs/source/integrations/triton.rst

```diff
@@ -1,3 +1,536 @@
+
+By default, TritonRunner uses gRPC protocol that is defined in Inference protocol [#triton_inference_protocol]_.
+
+HTTP/REST APIs is disabled by default, though it can be enabled via ``BENTOML_USE_HTTP_TRITONSERVER`` environment variable.
+
+.. code-block:: bash
+
+    $ docker run --init --rm -p 3000:3000 triton-integration:cpu serve \
+                            --production --triton-options model-control-mode=explicit \
+                            --triton-options load-model=onnx_mnist --triton-options load-model=torchscript_yolov5s
+
+.. epigraph::
+    Currently, TritonRunner does not support running `Metrics server <https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/user_guide/metrics.html>`_.
+    If you are interested in supporting the metrics server, please open an issue on :github:`GitHub <bentoml/BentoML/issues/new/choose>`
+
+Additionally, BentoML supervisors will allocate a random port for the gRPC server, hence ``grpc-port`` options that is passed to ``--triton-options`` will be omited.
+
+---------------
+
+Remarks
+~~~~~~~
+
+It is worth to mention that Triton Inference Server is a highly optimized model server that aims simplify multi-models serving.
+The following list of benefits and drawbacks are based on our experience building this integration:
+
+:raw-html:`<br />`
+
+The benefits Triton Inference Server brings:
+
+* **Performance enhancement**: Triton Inference Server's concurrent model execution [#concurrent_model_execution]_
+  enables models to be executed concurrently on the same/multiple GPUs, with out the IO performance limitations introduced by Python's GIL [#constraints]_.
+
+* **Extensive configuration options**: To go along with its suite of features and optimizations, Triton Inference Server also provides
+  an extensive set of configuration options that can be used to customize models' behaviour, with ability to extend its capabilities.
+
+* **Framework supports and custom backends**: Triton supports all major frameworks, including Tensorflow, PyTorch, ONNX, TensorRT [#tensorrt_support]_ and OpenVINO,
+  which enables for a wide range of use-cases and empowers ML practitioners to do what they do best. Additionally, they also support custom backends
+  that can be used to implement custom inference logic.
+
+
+Should I use Triton Inference Server?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Absolutely, depending on your use-case. If you are looking for improving RPS for your production models, then Triton Inference Server through ``bentoml.triton.Runner``
+is a great option to consider. Note that you are responsible for managing the model repository and its respectively ``config.pbtxt`` to make sure it will fit your use-case.
+
+If you have a lot of models to manage, then we recommend to only use TritonRunner for larger models that requires more resources to run inference, and use BentoML's
+:ref:`Runner <concepts/runner:Using Runners>` for smaller models.
```

Comment:
The recommendation is not necessarily divided by big and small models. Triton is ideal when then load is high and serving efficiency needs to be maximized.